website/src/parsers/glsl/glsl-parser.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import defaultParserInterface from '../utils/defaultParserInterface';
+import pkg from 'glsl-parser/package.json';
+import SettingsRenderer from '../utils/SettingsRenderer';
+
+const ID = 'glsl-parser';
+
+export default {
+  ...defaultParserInterface,
+
+  id: ID,
+  displayName: ID,
+  version: pkg.version,
+  homepage: pkg.homepage,
+  locationProps: new Set(['loc']),
+  _ignoredProperties: new Set(['loc']), // we ignore the loc itself because it's actually a locally enhanced (not in the actual parser data)
+
+  loadParser(callback) {
+    require(['glsl-tokenizer/string', 'glsl-parser/direct'], (
+      tokenize,
+      parse
+    ) => {
+      callback({ tokenize, parse });
+    });
+  },
+
+  parse({ tokenize, parse }, code) {
+    const tokens = tokenize(code);
+    const ast = parse(tokens);
+    // the parser does not yet provide the "end" so this is a workaround https://github.com/stackgl/glsl-parser/issues/17
+    function decoratePosition(node, end) {
+      node.loc = {
+        start: node.token.position || 0,
+        end,
+      };
+      node.children.forEach((child, i) => {
+        const nextSibling = node.children[i + 1];
+        decoratePosition(
+          child,
+          nextSibling && nextSibling.token && 'position' in nextSibling.token
+            ? nextSibling.token.position -
+                (nextSibling.token.preceding || [])
+                  .reduce((s, n) => s + (n.data || '').length, 0)
+            : end
+        );
+      });
+    }
+    decoratePosition(ast, code.length);
+    return ast;
+  },
+
+  nodeToRange({ loc }) {
+    if (loc) {
+      return [loc.start, loc.end];
+    }
+  },
+
+  getNodeName(node) {
+    return node.type;
+  },
+
+  opensByDefault(node, key) {
+    return key === 'children' && node.type === '(program)';
+  },
+
+  renderSettings(parserSettings, onChange) {
```

Comment:
There is no need to add this method if there are no settings. In fact, I think it should not be there so that the button is disabled in the UI.